test/test_tensorboard.py

```diff
@@ -802,11 +804,16 @@ def test_scalar(self):
         self.assertIsInstance(res, np.ndarray) and self.assertEqual(res.shape, (1,))
         res = make_np(np.float16(1.00000087))
         self.assertIsInstance(res, np.ndarray) and self.assertEqual(res.shape, (1,))
-        res = make_np(np.float128(1.00008 + 9))
-        self.assertIsInstance(res, np.ndarray) and self.assertEqual(res.shape, (1,))
         res = make_np(np.int64(100000000000))
         self.assertIsInstance(res, np.ndarray) and self.assertEqual(res.shape, (1,))
 
+    @skipIfTorchDynamo
```

Comment:
We can probably fix this one rather easily.

Reply:
No, the issue here is that we should break on `float128`s and we are not doing so atm.

Reply:
We are breaking on np.float128.
Running

```
(pytorch-dev) ev-br@qgpu3:~/repos/pytorch$ cat ma.py 
import torch
from torch._dynamo import config
import torch._dynamo.testing
import numpy as np


def fn(x):
    return np.float128(x)

print(fn(1e100))


cnts = torch._dynamo.testing.CompileCounter()
opt_fn = torch._dynamo.optimize(cnts)(fn)

print(opt_fn(1e100))
print(cnts.frame_count)
```

produces

```
(pytorch-dev) ev-br@qgpu3:~/repos/pytorch$ TORCH_LOGS="graph_breaks" python ma.py 
1.0000000000000000159e+100
[2023-10-04 14:09:17,525] [0/0] torch._dynamo.symbolic_convert.__graph_breaks: [DEBUG] Graph break: Can't find numpy function <class 'numpy.float128'> in torch._numpy.  Please file an issue to request support for this function. from user code at:
[2023-10-04 14:09:17,525] [0/0] torch._dynamo.symbolic_convert.__graph_breaks: [DEBUG]   File "ma.py", line 29, in fn
[2023-10-04 14:09:17,525] [0/0] torch._dynamo.symbolic_convert.__graph_breaks: [DEBUG]     return np.float128(x)
[2023-10-04 14:09:17,525] [0/0] torch._dynamo.symbolic_convert.__graph_breaks: [DEBUG] 
1.0000000000000000159e+100
0
```

Reply:
Okay, this is a bit trickier. The example above is correct, but. If `np.float128(42)` is used as input to a function, it was not recognized as a numpy variable and eventually there was a hard failure at https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/variables/builder.py#L1230.

One solution, which this PR does, is to make `is_numpy` function aware of array scalars and wrap them on an equal footing as ndarrays. This is what `torch._numpy` assumes anyway, so it should it be safe I expect.

Reply:
Wrapping NumPy scalars into arrays is good for now. Something we should do in the future is to implement proper scalar semantics on NumPy scalars, even if internally we represent them as a tensor.

One day...

Reply:
Discussed this offline with Mario, and the current state of the PR does not skip this test anymore (was a bit of a rabbit hole but hey).